### DISCUSSION TITLE
CLusters?

### DISCUSSION THREAD

**Discussion by DevMan11** [JellyChain1/JellyChain1#6] [Ideas]
_**I separated the Clusters and Zero Downtime servers. The Zero Downtime server is the same as the Clusters. I tested it several times they passed 100%. The No-Kill server is built-in it has no stop/kill function that will fork the worker process if it is killed. This loop will cause zero downtime. When can we make the source code public for the community to work on.**_

**Comment by Cotopia-Inc** [comment]
We are working on finalizing a few things and it should be public this week.

**Reply by DevMan11** [reply]
If it's something I can do let me know.

**Reply by Cotopia-Inc** [reply]
Thank You

**Comment by DevMan11** [comment]
So this new Clusters module will fork the processes on your CPU. Will this give me multiple ports?

**Reply by HeartBreakKid8** [reply]
yes, when I tested the module it made more ports for me to use. What happens is the Clusters module will take your processor give it a master port number and the fork the master port number into multiple ports. So if your CPU have 4 processors it will make one master into 4 worker ports.

**Reply by Cotopia-Inc** [reply]
Well said

**Comment by Cotopia-Inc** [comment]
The new Zero Downtime cluster server source codes are ready for public review. Please let us know what you think. We want to hear feedback from the community. Thank You

**Comment by BotGame22** [comment]
Working with this Clusters Module is interesting. It takes my master port and make me 4 extra ports that I used to connect other apps to the chain. The best part is I just point my rpc server and the blockchain server to the worker ports and I was rolling. This is a pretty good Idea.

**Reply by Cotopia-Inc** [reply]
Thank You please let us know about any other experiences you may have.

**Comment by CrestTaffy** [comment]
I was able to run a node.js process on one port and a python process on another and a completely different application on the rest. Adding Clusters to the Zero Downtime Server was a great Idea.
